GO language build break

Hi , there is a build break for GO language server . 

usr/local/go-packages/pkg/mod/github.com/golangci/golangci-lint@v1.43.0/internal/robustio/robustio_other.go:18:9: undefined: os.ReadFile
note: module requires Go 1.16
 github.com/golangci/golangci-lint/pkg/fsutils
usr/local/go-packages/pkg/mod/github.com/golangci/golangci-lint@v1.43.0/pkg/fsutils/filecache.go:27:20: undefined: os.ReadFile
note: module requires Go 1.16
 github.com/go-critic/go-critic/checkers
usr/local/go-packages/pkg/mod/github.com/go-critic/go-critic@v0.6.1/checkers/ruleguard_checker.go:149:17: undefined: os.ReadFile
 github.com/breml/bidichk/pkg/bidichk
usr/local/go-packages/pkg/mod/github.com/breml/bidichk@v0.1.1/pkg/bidichk/bidichk.go:46:15: undefined: os.ReadFile
note: module requires Go 1.16
The command '/bin/sh -c go get -u -v github.com/mdempsky/gocode &&     go get -u -v github.com/uudashr/gopkgs/cmd/gopkgs &&     go get -u -v github.com/ramya-rao-a/go-outline &&     go get -u -v github.com/acroca/go-symbols &&     go get -u -v golang.org/x/tools/cmd/guru &&     go get -u -v golang.org/x/tools/cmd/gorename &&     go get -u -v github.com/fatih/gomodifytags &&     go get -u -v github.com/haya14busa/goplay/cmd/goplay &&     go get -u -v github.com/josharian/impl &&     go get -u -v github.com/tylerb/gotype-live &&     go get -u -v github.com/rogpeppe/godef &&     go get -u -v github.com/zmb3/gogetdoc &&     go get -u -v golang.org/x/tools/cmd/goimports &&     go get -u -v github.com/sqs/goreturns &&     go get -u -v winterdrache.de/goformat/goformat &&     go get -u -v golang.org/x/lint/golint &&     go get -u -v github.com/cweill/gotests/... &&     go get -u -v github.com/alecthomas/gometalinter &&     go get -u -v honnef.co/go/tools/... &&     GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint &&     go get -u -v github.com/mgechev/revive &&     go get -u -v github.com/sourcegraph/go-langserver &&     go get -u -v github.com/go-delve/delve/cmd/dlv &&     go get -u -v github.com/davidrjenni/reftools/cmd/fillstruct &&     go get -u -v github.com/godoctor/godoctor' returned a non-zero code: 2


Changing the ENV GO_VERSION=1.15 \     doesnt to  1.16 doesn't work either . and below is the error when changed the version to 1.16 : 

 ---> Running in 1dbce08b9671
go: downloading github.com/mdempsky/gocode v0.0.0-20200405233807-4acdcbdea79d
go: downloading golang.org/x/tools v0.1.7
golang.org/x/tools/internal/typeparams
github.com/mdempsky/gocode/internal/lookdot
golang.org/x/tools/go/internal/gcimporter
github.com/mdempsky/gocode/internal/suggest
golang.org/x/tools/go/gcexportdata
github.com/mdempsky/gocode/internal/cache
github.com/mdempsky/gocode/internal/gbimporter
github.com/mdempsky/gocode
go: downloading github.com/uudashr/gopkgs v1.3.2
go: downloading github.com/uudashr/gopkgs v2.0.1+incompatible
go: downloading github.com/uudashr/gopkgs/cmd/gopkgs v0.0.0-20191024034442-58e9141cd7d6
go get: github.com/uudashr/gopkgs/cmd/gopkgs@none updating to
	github.com/uudashr/gopkgs/cmd/gopkgs@v0.0.0-20191024034442-58e9141cd7d6 requires
	github.com/uudashr/gopkgs/v2@v2.1.0 requires
	github.com/uudashr/gopkgs@v0.0.0: reading github.com/uudashr/gopkgs/go.mod at revision v0.0.0: unknown revision v0.0.0
The command '/bin/sh -c go get -u -v github.com/mdempsky/gocode &&     go get -u -v github.com/uudashr/gopkgs/cmd/gopkgs &&     go get -u -v github.com/ramya-rao-a/go-outline &&     go get -u -v github.com/acroca/go-symbols &&     go get -u -v golang.org/x/tools/cmd/guru &&     go get -u -v golang.org/x/tools/cmd/gorename &&     go get -u -v github.com/fatih/gomodifytags &&     go get -u -v github.com/haya14busa/goplay/cmd/goplay &&     go get -u -v github.com/josharian/impl &&     go get -u -v github.com/tylerb/gotype-live &&     go get -u -v github.com/rogpeppe/godef &&     go get -u -v github.com/zmb3/gogetdoc &&     go get -u -v golang.org/x/tools/cmd/goimports &&     go get -u -v github.com/sqs/goreturns &&     go get -u -v winterdrache.de/goformat/goformat &&     go get -u -v golang.org/x/lint/golint &&     go get -u -v github.com/cweill/gotests/... &&     go get -u -v github.com/alecthomas/gometalinter &&     go get -u -v honnef.co/go/tools/... &&     GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint &&     go get -u -v github.com/mgechev/revive &&     go get -u -v github.com/sourcegraph/go-langserver &&     go get -u -v github.com/go-delve/delve/cmd/dlv &&     go get -u -v github.com/davidrjenni/reftools/cmd/fillstruct &&     go get -u -v github.com/godoctor/godoctor' returned a non-zero code: 1


I could root cause the issue is from " go get -u -v honnef.co/go/tools/... && \ " command  =>  https://github.com/golang/tools

--- a/theia-full-docker/Dockerfile
+++ b/theia-full-docker/Dockerfile
@@ -127,46 +127,28 @@ FROM common
 
 # LSPs
 
-## GO
-ENV GO_VERSION=1.15 \
+## Go
+ENV GO_VERSION=1.17.3 \
     GOOS=linux \
     GOARCH=amd64 \
     GOROOT=/usr/local/go \
     GOPATH=/usr/local/go-packages
-ENV PATH=$GOPATH/bin:$GOROOT/bin:$PATH
+ENV PATH=$GOROOT/bin:$GOPATH/bin:$PATH
 
 # Install Go
+# https://go.dev/doc/install
 RUN curl -fsSL https://storage.googleapis.com/golang/go$GO_VERSION.$GOOS-$GOARCH.tar.gz | tar -C /usr/local -xzv
 
-# Install VS Code Go tools: https://github.com/Microsoft/vscode-go/blob/058eccf17f1b0eebd607581591828531d768b98e/src/goInstallTools.ts#L19-L45
-RUN go get -u -v github.com/mdempsky/gocode && \
-    go get -u -v github.com/uudashr/gopkgs/cmd/gopkgs && \
+# VS Code Go Tools https://github.com/golang/vscode-go/blob/master/docs/tools.md
+RUN go get -u -v github.com/uudashr/gopkgs/cmd/gopkgs@v2 && \
     go get -u -v github.com/ramya-rao-a/go-outline && \
-    go get -u -v github.com/acroca/go-symbols && \
-    go get -u -v golang.org/x/tools/cmd/guru && \
-    go get -u -v golang.org/x/tools/cmd/gorename && \
+    go get -u -v github.com/cweill/gotests/gotests && \
     go get -u -v github.com/fatih/gomodifytags && \
-    go get -u -v github.com/haya14busa/goplay/cmd/goplay && \
     go get -u -v github.com/josharian/impl && \
-    go get -u -v github.com/tylerb/gotype-live && \
-    go get -u -v github.com/rogpeppe/godef && \
-    go get -u -v github.com/zmb3/gogetdoc && \
-    go get -u -v golang.org/x/tools/cmd/goimports && \
-    go get -u -v github.com/sqs/goreturns && \
-    go get -u -v winterdrache.de/goformat/goformat && \
-    go get -u -v golang.org/x/lint/golint && \
-    go get -u -v github.com/cweill/gotests/... && \
-    go get -u -v github.com/alecthomas/gometalinter && \
-    go get -u -v honnef.co/go/tools/... && \
-    GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint && \
-    go get -u -v github.com/mgechev/revive && \
-    go get -u -v github.com/sourcegraph/go-langserver && \
+    go get -u -v github.com/haya14busa/goplay/cmd/goplay && \
     go get -u -v github.com/go-delve/delve/cmd/dlv && \
-    go get -u -v github.com/davidrjenni/reftools/cmd/fillstruct && \
-    go get -u -v github.com/godoctor/godoctor
-
-RUN go get -u -v -d github.com/stamblerre/gocode && \
-    go build -o $GOPATH/bin/gocode-gomod github.com/stamblerre/gocode
+    GO111MODULE=on go get -v github.com/golangci/golangci-lint/cmd/golangci-lint && \
+                   go get -u -v golang.org/x/tools/gopls@v0.7.3
 
 ENV PATH=$PATH:$GOPATH/bin
 
@@ -176,6 +158,11 @@ RUN apt-get update && apt-get -y install openjdk-11-jdk maven gradle
 # C/C++
 # public LLVM PPA, development version of LLVM
 ARG LLVM=14
+
+RUN apt-get update && \
+    apt-get install -y software-properties-common && \
+    add-apt-repository ppa:ubuntu-toolchain-r/test && \
+    apt-get remove -y software-properties-common
 
 RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - && \
     echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic main" > /etc/apt/sources.list.d/llvm.list && \

--- a/theia-go-docker/Dockerfile
+++ b/theia-go-docker/Dockerfile
@@ -42,45 +42,28 @@ RUN adduser --disabled-password --gecos '' theia && \
 
 USER theia
 
-ENV GO_VERSION=1.15 \
+## Go
+ENV GO_VERSION=1.17.3 \
     GOOS=linux \
     GOARCH=amd64 \
-    GOROOT=/home/go \
-    GOPATH=/home/go-tools
-ENV PATH=$GOPATH/bin:$GOROOT/bin:$PATH
+    GOROOT=/usr/local/go \
+    GOPATH=/usr/local/go-packages
+ENV PATH=$GOROOT/bin:$GOPATH/bin:$PATH
 
 # Install Go
-RUN curl -fsSL https://storage.googleapis.com/golang/go$GO_VERSION.$GOOS-$GOARCH.tar.gz | tar -C /home -xzv
+# https://go.dev/doc/install
+RUN curl -fsSL https://storage.googleapis.com/golang/go$GO_VERSION.$GOOS-$GOARCH.tar.gz | tar -C /usr/local -xzv
 
-# Install VS Code Go tools: https://github.com/Microsoft/vscode-go/blob/058eccf17f1b0eebd607581591828531d768b98e/src/goInstallTools.ts#L19-L45
-RUN go get -u -v github.com/mdempsky/gocode && \
-    go get -u -v github.com/uudashr/gopkgs/cmd/gopkgs && \
+# VS Code Go Tools https://github.com/golang/vscode-go/blob/master/docs/tools.md
+RUN go get -u -v github.com/uudashr/gopkgs/cmd/gopkgs@v2 && \
     go get -u -v github.com/ramya-rao-a/go-outline && \
-    go get -u -v github.com/acroca/go-symbols && \
-    go get -u -v golang.org/x/tools/cmd/guru && \
-    go get -u -v golang.org/x/tools/cmd/gorename && \
+    go get -u -v github.com/cweill/gotests/gotests && \
     go get -u -v github.com/fatih/gomodifytags && \
-    go get -u -v github.com/haya14busa/goplay/cmd/goplay && \
     go get -u -v github.com/josharian/impl && \
-    go get -u -v github.com/tylerb/gotype-live && \
-    go get -u -v github.com/rogpeppe/godef && \
-    go get -u -v github.com/zmb3/gogetdoc && \
-    go get -u -v golang.org/x/tools/cmd/goimports && \
-    go get -u -v github.com/sqs/goreturns && \
-    go get -u -v winterdrache.de/goformat/goformat && \
-    go get -u -v golang.org/x/lint/golint && \
-    go get -u -v github.com/cweill/gotests/... && \
-    go get -u -v github.com/alecthomas/gometalinter && \
-    go get -u -v honnef.co/go/tools/... && \
-    GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint && \
-    go get -u -v github.com/mgechev/revive && \
-    go get -u -v github.com/sourcegraph/go-langserver && \
+    go get -u -v github.com/haya14busa/goplay/cmd/goplay && \
     go get -u -v github.com/go-delve/delve/cmd/dlv && \
-    go get -u -v github.com/davidrjenni/reftools/cmd/fillstruct && \
-    go get -u -v github.com/godoctor/godoctor
-
-RUN go get -u -v -d github.com/stamblerre/gocode && \
-    go build -o $GOPATH/bin/gocode-gomod github.com/stamblerre/gocode
+    GO111MODULE=on go get -v github.com/golangci/golangci-lint/cmd/golangci-lint && \
+                   go get -u -v golang.org/x/tools/gopls@v0.7.3
 
 # Configure Theia
 ENV SHELL=/bin/bash \

